src/Chapter3.hs

```diff
@@ -480,6 +513,51 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Shrine = Church | Library deriving (Show)
+
+data City = City
+    { cityCastle  :: Maybe Castle
+    , cityWall    :: Maybe Wall
+    , cityShrine  :: Shrine
+    , cityHouses  :: [House]
+    } deriving (Show)
+
+data Castle = Castle
+    { castleName :: String
+    } deriving (Show)
+
+data Wall = Wall
+    { wallHeight :: Int
+    } deriving (Show)
+
+data House = House
+    { houseRoom :: Int
+    } deriving (Show)
+
+mkHouse :: Int -> Maybe House
+mkHouse rooms
+    | rooms > 4 = Nothing
+    | otherwise = Just (House rooms)
```

Comment:
Wonderful solutions! If you're curious, this design pattern is usually called "smart constructor".
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Also, you probably want to add check if `rooms < 1`